### PR TITLE
Document `linea_getProof`

### PR DESCRIPTION
**Pull request by bgravenorst**
Document the `linea_getProof` method. 
Updated method structure to be consistent with MetaMask

### CODE REVIEW

File: docs/developers/reference/api/linea-getproof.mdx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+  - `key`: The address of the account for which the proof is generated.
+  - `leafIndex`: The index of the leaf node in the Merkle Patricia Trie that corresponds to the account.
+  - `proof`: Object containing the proof data for the account:
+    - `proofRelatedNodes`: Array of nodes in the Merkle Patricia Trie used to construct the proof.
+    - `value`: The value stored at the account's leaf node in the Merkle Patricia Trie.
+- `storageProof`: An array of storage-entries as requested. Each entry is an object with these properties:
+  - `key`: The requested storage key.
+  - `leftLeafIndex`: The index of the left leaf node in the Merkle Patricia Trie that corresponds to the
+      storage slot.
+  - `leftProof`: Contains the proof data for the left leaf node, which inlcudes the array of nodes used
```
Comment (dfox-consensys) [inline docs/developers/reference/api/linea-getproof.mdx:41]: includes

File: docs/developers/reference/api/linea-getproof.mdx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      to construct the proof and the value stored at the account's leaf node in the Merkle Patricia Trie.
+  - `rightLeafIndex`: The index of the right leaf node in the Merkle Patricia Trie that corresponds to
+      the storage slot.
+  - `rightProof`: Contains the proof data for the right leaf node, which inlcudes the array of nodes used
```
Comment (dfox-consensys) [inline docs/developers/reference/api/linea-getproof.mdx:45]: includes

File: docs/developers/reference/api/linea-getproof.mdx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+Linea currently does not support the block parameter `finalized`, meaning that to get the current
+finalised block on the L1, users must retrieve the value from the `currentL2BlockNumber` variable in
+the [LineaRollup L1 contract](https://etherscan.io/address/0xd19d4b5d358258f05d7b411e21a1460d11b0876f#readProxyContract).
+:::
+
+## Parameters
+
+- `address`: A string representing the address (20 bytes) to check for balance.
+- `storageKeys`: An array of 32-byte storage keys to be proofed and included.
+- `blockParameter`: A hexadecimal block number, or one of the string tags `latest`, `earliest`, or
```
Comment (dfox-consensys) [inline docs/developers/reference/api/linea-getproof.mdx:26]: Should we say this is optional?

File: docs/developers/reference/api/linea-getproof.mdx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+:::info important
+
+Linea currently does not support the block parameter `finalized`, meaning that to get the current
+finalised block on the L1, users must retrieve the value from the `currentL2BlockNumber` variable in
+the [LineaRollup L1 contract](https://etherscan.io/address/0xd19d4b5d358258f05d7b411e21a1460d11b0876f#readProxyContract).
+:::
+
+## Parameters
+
+- `address`: A string representing the address (20 bytes) to check for balance.
```
Comment (jonesho) [inline docs/developers/reference/api/linea-getproof.mdx:24]: `to check for balance` seems too restricted for the use of account balance check (a proof can verify other account values like nonce, codehash, etc..), maybe we could say in a more general way like `for which the proof is requested`?

File: docs/developers/reference/api/linea-getproof.mdx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+  - `leafIndex`: The index of the leaf node in the Merkle Patricia Trie that corresponds to the account.
+  - `proof`: Object containing the proof data for the account:
+    - `proofRelatedNodes`: Array of nodes in the Merkle Patricia Trie used to construct the proof.
+    - `value`: The value stored at the account's leaf node in the Merkle Patricia Trie.
+- `storageProof`: An array of storage-entries as requested. Each entry is an object with these properties:
+  - `key`: The requested storage key.
+  - `leftLeafIndex`: The index of the left leaf node in the Merkle Patricia Trie that corresponds to the
+      storage slot.
+  - `leftProof`: Contains the proof data for the left leaf node, which includes the array of nodes used
+      to construct the proof and the value stored at the account's leaf node in the Merkle Patricia Trie.
```
Comment (jonesho) [inline docs/developers/reference/api/linea-getproof.mdx:42]: `stored at the account's leaf node` should be `stored at the storage's leaf node`?

File: docs/developers/reference/api/linea-getproof.mdx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+  - `rightLeafIndex`: The index of the right leaf node in the Merkle Patricia Trie that corresponds to
+      the storage slot.
+  - `rightProof`: Contains the proof data for the right leaf node, which includes the array of nodes used
+      to construct the proof and the value stored at the account's leaf node in the Merkle Patricia Trie.
```
Comment (jonesho) [inline docs/developers/reference/api/linea-getproof.mdx:46]: Same here, `stored at the account's leaf node` should be `stored at the right leaf node`?

**Comment by matkt** [comment]
For linea we are using merkle sparse tree . So I will replace MPT by Merkle sparse tree

**Comment by matkt** [comment]
It's a bit more complicated than that. When an account or storage exists, we will return the leaf index and a proof of existence for it. If the account or storage does not exist, we will return the proof of the element before and after it in the tree. So, the returned JSON can be different regarding the element exist or not. For example, in your documentation, the account exists but the slot does not. 

https://github.com/Consensys/shomei/tree/main/trie/src/main/java/net/consensys/shomei/trie/proof

File: docs/developers/reference/api/eth-sendrawtransaction.mdx [hunk 7]
```diff
@@ -46,8 +46,10 @@ The 32-byte transaction hash, or the zero hash if the transaction is not yet ava
 
 ## Example
 
+### Request
+
 <Tabs>
```
Comment (jlwllmr) [inline docs/developers/reference/api/eth-sendrawtransaction.mdx:51]: I'm fine with diving up the Example section into Request and Response, but it doesn't seem to make sense to leave the tabs in. Currently we only have one tab in each `<Tabs>`?

File: docs/developers/reference/api/linea-estimategas.mdx [hunk 8]
```diff
@@ -86,40 +88,44 @@ You can also call the API using [Infura's supported Linea endpoints](https://doc
...
+
+    const fees: LineaEstimateGasResponse = await provider.send("linea_estimateGas", [params]);
+    console.log(fees);
+    ``` 
+  </TabItem>
+</Tabs>
+
+### Response
+
+  <Tabs>
```
Comment (jlwllmr) [inline docs/developers/reference/api/linea-estimategas.mdx:117]: Again, do we still need the `<Tabs>` if there's only one tab?

**Review by jlwllmr — APPROVED** [review]

**Pull request merged**